### DISCUSSION TITLE
Uable to query transactions of IBC signer

### DISCUSSION THREAD

**Discussion by Fafnir12321** [celestiaorg/celestia-node#3510] [Q&A]
I ' m using TIA ' s public node and self - hosted node , but neither node is able to look up the transactions for **transfer.sender** for **celestia16p6Irixf7f03c0ka8cv4sznr29rym27us7u4v6** which is an IBC signer. I can see the transaction history of this account and the balance of the account on Celestia explore , how should I achieve this using self-hosted node?

Here are the situations for public node and self-hosted node.
1. curl for public node:

curl -- location 'https://api.celestia.pops.one/cosmos/tx/v1beta1/txs?limit=100&page=0&events=transfer.sender%3D%27celestia16p6lrlxf7f03c0ka8cv4sznr29rym27us7u4v6%27'

Response:
![IMG_20240619_022023](https://github.com/celestiaorg/celestia-node/assets/173134836/2440bc5b-e7bb-40e7-9e48-be25d711a6c5)

2. When using same api to query self-hosted node, the self-hosted node will query data, then it will crash and run out of memory.

**Comment by walldiss** [comment]
Hello, thank you for the report!

Listed endpoint is public and may see some internal problems. You are welcome to use any other from list of public da endpoints listed in https://docs.celestia.org/nodes/mainnet

Could you please provide more info on what type of node you are running self hosted:
- type of the node
- Memory of the machine
- could you please send logs produced by node when it crashes.

**Reply by Fafnir12321** [reply]
Thanks for your reply.
1. The curl for public node I mentioned in the question is one of the nodes in the document you listed.
2. About self-host node:

-  Full storage node
-  32G
-  there were no error logs and it restarts directly after executing the query for some time. But I did check the memoru usage and found it was out of memory. I got a response: curl:(52) Empty reply from server.

Currently, this looks like it can be solved by adding memory, but I'm still have these questions. 
- Why it takes so much memory to query this address, simply because of the huge number of transactions? 
- How much memory is needed go successfully query this address? 
- Is there a better way to solve this problem?
